sandbox/modules/identity/src/main/java/org/opensearch/identity/IdentityPlugin.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.identity;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.action.support.ActionFilter;
+import org.opensearch.authn.AuthenticationManager;
+import org.opensearch.authn.internal.InternalAuthenticationManager;
+import org.opensearch.client.Client;
+import org.opensearch.cluster.metadata.IndexNameExpressionResolver;
+import org.opensearch.cluster.service.ClusterService;
+import org.opensearch.common.io.stream.NamedWriteableRegistry;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.common.util.concurrent.ThreadContext;
+import org.opensearch.common.xcontent.NamedXContentRegistry;
+import org.opensearch.env.Environment;
+import org.opensearch.env.NodeEnvironment;
+import org.opensearch.plugins.ActionPlugin;
+import org.opensearch.plugins.NetworkPlugin;
+import org.opensearch.plugins.Plugin;
+import org.opensearch.repositories.RepositoriesService;
+import org.opensearch.rest.RestHandler;
+import org.opensearch.script.ScriptService;
+import org.opensearch.threadpool.ThreadPool;
+import org.opensearch.watcher.ResourceWatcherService;
+
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Objects;
+import java.util.function.Supplier;
+import java.util.function.UnaryOperator;
+
+public final class IdentityPlugin extends Plugin implements ActionPlugin, NetworkPlugin {
+    private volatile Logger log = LogManager.getLogger(this.getClass());
+
+    private volatile SecurityRestFilter securityRestHandler;
+    private volatile Settings settings;
+
+    private volatile Path configPath;
+    private volatile SecurityFilter sf;
+    private volatile ThreadPool threadPool;
+    private volatile ClusterService cs;
+    private volatile Client localClient;
+    private volatile NamedXContentRegistry namedXContentRegistry = null;
+
+    @SuppressWarnings("removal")
+    public IdentityPlugin(final Settings settings, final Path configPath) {
+        this.configPath = configPath;
+
+        if (this.configPath != null) {
+            log.info("OpenSearch Config path is {}", this.configPath.toAbsolutePath());
+        } else {
+            log.info("OpenSearch Config path is not set");
+        }
+
+        this.settings = settings;
+    }
+
+    @Override
+    public UnaryOperator<RestHandler> getRestHandlerWrapper(final ThreadContext threadContext) {
```

Comment:
Any issues running side-by-side with the security plugin?  If so we should document so we can figure out how to work together

Reply:
Yes, this still enforces that only one plugin registers this extension point, but I plan to address that in a follow-up PR. The identity module will always register this extension point and the check will allow only one other plugin to register it.